sdk/core/Azure.Core.Experimental/src/Serialization/JsonObjectSerializer.cs

```diff
@@ -59,5 +67,85 @@ public override async ValueTask<object> DeserializeAsync(Stream stream, Type ret
         {
             return await JsonSerializer.DeserializeAsync(stream, returnType, _options, cancellationToken).ConfigureAwait(false);
         }
+
+        /// <inheritdoc/>
+        public string? GetSerializedMemberName(MemberInfo memberInfo)
+        {
+            Argument.AssertNotNull(memberInfo, nameof(memberInfo));
+
+            if (!_cache.TryGetValue(memberInfo.ReflectedType, out MemberInfo[] members))
+            {
+                members = GetMembers(memberInfo.ReflectedType).ToArray();
```

Comment:
Should we turn this into a `Dictionary<MemberInfo, string>` for faster lookup?

Reply:
STJ has code to handle cases where different members can serialize the same in polymorphic ways. This mimics that to more closely match STJ's implementation, which is necessary to better ensure we get the same information that STJ will serialize.

Reply:
To note, this implementation - per the comments with links to the code - is written almost verbatim with STJ, eliding parts of the code we don't need for just mapping the name.

Reply:
But do those polymorphic ways change per invocation?  If I look something up using the same MemberInfo with the same serialization settings, won't it always give me back the same serialized name?  Maybe I missed that part.